packages/dds/tree/src/test/utils.ts

```diff
@@ -26,3 +39,110 @@ export class MockDependent extends SimpleObservingDependent {
+     * @param trees - the number of trees to initialize this provider with. This is the same as calling
+     * {@link create} followed by {@link createTree} _trees_ times.
+     *
+     * @example
+     * ```ts
+     * const trees = await TestTreeProvider.createTrees(2);
+     * assert(trees[0].isAttached());
+     * assert(trees[1].isAttached());
+     * await trees.ensureSynchronized();
+     * ```
+     */
+    public static async createTrees(trees: number): Promise<TestTreeProvider & ITestObjectProvider> {
+        const provider = await this.create();
+        for (let i = 0; i < trees; i++) {
+            await provider.createTree();
+        }
+        return provider;
+    }
+
+    /**
+     * Create and initialize a new {@link SharedTree} that is connected to all other trees from this provider.
+     * @returns the tree that was created. For convenience, the tree can also be accessed via `this[i]` where
+     * _i_ is the index of the tree in order of creation.
+     */
+    public async createTree(): Promise<SharedTree> {
+        return this.createTreeAtIndex(this.trees.length);
+    }
+
+    public [Symbol.iterator](): IterableIterator<SharedTree> {
+        return this.trees[Symbol.iterator]();
+    }
+
+    private constructor() {
+        const factory = new SharedTreeFactory();
+        const registry = [[TestTreeProvider.treeId, factory]] as ChannelFactoryRegistry;
+        const driver = new LocalServerTestDriver();
+        this.provider = new TestObjectProvider(
+            Loader,
+            driver,
+            () => new TestContainerRuntimeFactory(
+                "@fluid-example/test-dataStore",
+                new TestFluidObjectFactory(registry),
+            ),
+        );
+
+        return new Proxy(this, {
+            get: (target, prop, receiver) => {
+                // Intercept numbers and retrieve the associated tree, e.g. `testTreeProvider[0]`;
+                if (typeof prop === "string") {
+                    const treeIndex = Number.parseInt(prop, 10);
+                    if (!Number.isNaN(treeIndex)) {
+                        assert(treeIndex < this.trees.length, `treeIndex out of bounds (${treeIndex})`);
+                        return this.trees[treeIndex];
+                    }
+                }
+
+                // Route all properties that are on the `TestTreeProvider` itself
+                if ((target as never)[prop] !== undefined) {
+                    return Reflect.get(target, prop, receiver) as unknown;
+                }
+
+                // Route all other properties to the `TestObjectProvider`
+                return Reflect.get(this.provider, prop, receiver) as unknown;
+            },
+        });
+    }
+
+    private async createTreeAtIndex(treeIndex: number): Promise<SharedTree> {
+        assert(treeIndex >= 0, `treeIndex out of bounds (${treeIndex})`);
+        if (treeIndex < this.trees.length) {
+            return this.trees[treeIndex];
```

Comment:
Returning preexisting/shared objects from a 'create' method is misleading.

Reply:
Good call, it used to be called twice and I forgot to smush it back together.